src/validation.cpp

```diff
@@ -4149,8 +4149,8 @@ std::vector<unsigned char> ChainstateManager::GenerateCoinbaseCommitment(CBlock&
 
 bool HasValidProofOfWork(const std::vector<CBlockHeader>& headers, const Consensus::Params& consensusParams)
 {
-    return std::all_of(headers.cbegin(), headers.cend(),
-            [&](const auto& header) { return CheckProofOfWork(header.GetHash(), header.nBits, consensusParams);});
+    return std::ranges::all_of(headers,
```

Comment:
nit: this change could either be merged with the [vector-to-std::span](https://github.com/bitcoin/bitcoin/pull/32740/commits/dac0af8fb03a7ba39a36b5f7fa204683ae146013) or moved over to that commit, it's not needed in 80c63c9ceccc89295f004099c8cd7e7dfd99b556